src/overlays/actors/ovl_En_Heishi2/z_en_heishi2.c

```diff
@@ -21,77 +67,880 @@ const ActorInit En_Heishi2_InitVars = {
+                    if (2 != this->unk_30A) {
+                        gate->isOpening = -1;
+                        break;
+                    }
+
+                    else {
+                        gate->isOpening = 2;
+                        break;
+                    }
+                }
+            } while (gate != NULL);
+        }
+        if (this->unk_30A == 0) {
+            this->unk_30A = 1;
+        }
+        this->actionFunc = func_80A53DF8;
+    }
+}
+void func_80A544AC(EnHeishi2* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.z, -0x17D4, 5, (s16)(s32)(f32)this->unk_2E4, 0);
+    Math_SmoothScaleMaxF(&this->unk_2E4, 3000.0f, 1.0f, 500.0f);
+    this->actor.posRot.rot.z = this->actor.shape.rot.z;
+    if (this->actor.shape.rot.z < -6000) {
+        func_8010B680(globalCtx, 0x708F, 0);
+        this->actor.flags |= 0x10000;
+        this->actionFunc = func_80A5455C;
+        this->unk_2E4 = 0.0f;
+    }
+}
+
+#ifdef NON_MATCHING
+// ordering and float stuff
+void func_80A5455C(EnHeishi2* this, GlobalContext* globalCtx) {
+    EnBom* bomb;
+    Vec3f vec;
+    Actor* thisx = &this->actor;
+    s16 temp;
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5)) {
+        if (func_80106BC8(globalCtx) != 0) {
+
+            func_8002DF54(globalCtx, NULL, 7);
+            func_80106CCC(globalCtx);
+            vec.x = Math_Rand_CenteredFloat(20.0f) + this->unk_274.x;
+            vec.y = Math_Rand_CenteredFloat(20.0f) + (this->unk_274.y - 40.0f);
+            vec.z = Math_Rand_CenteredFloat(20.0f) + (this->unk_274.z - 20.0f);
+            bomb = (EnBom*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, vec.x, vec.y, vec.z, 0,
+                                       (s16)(Math_Rand_CenteredFloat(7000.0f) + thisx->rotTowardsLinkY), 0, 0);
+
+            if (bomb != NULL) {
+                bomb->actor.speedXZ = Math_Rand_CenteredFloat(5.0f) + 10.0f;
+                bomb->actor.velocity.y = Math_Rand_CenteredFloat(5.0f) + 10.0f;
+            }
+            // This is down!
+            osSyncPrintf(VT_FGCOL(YELLOW) " ☆☆☆☆☆ これでダウンだ！ ☆☆☆☆☆ \n" VT_RST);
+            this->actionFunc = func_80A546DC;
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi2/func_80A5455C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi2/func_80A546DC.s")
-
+#endif
+
+void func_80A546DC(EnHeishi2* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+    Math_SmoothScaleMaxMinS(&thisx->shape.rot.z, 200, 5, this->unk_2E4, 0);
+    Math_SmoothScaleMaxF(&this->unk_2E4, 3000.0f, 1.0f, 500.0f);
+    thisx->posRot.rot.z = thisx->shape.rot.z;
+    if (thisx->shape.rot.z > 0) {
+        Actor_Kill(thisx);
+    }
+}
+
+#ifdef NON_MATCHING
+// can't get this to match, i think it's mostly functionally equivalent
+void func_80A5475C(EnHeishi2* this, GlobalContext* globalCtx) {
+    Actor* thisx;
+    s16 angleDiff;
+    s16 newAngleDiff;
```

Comment:
this one can use elses and format fixes. I wont comment on all of them though.
I would try to make extra sure that its functionally equivalent though, we dont want to use NON_MATCHING on it otherwise